include/openmc/particle_data.h

```diff
@@ -229,163 +375,112 @@ struct BoundaryInfo {
-  int g_last_;     //!< pre-collision energy group (MG only)
-
-  // Other physical data
-  double wgt_ {1.0};       //!< particle weight
-  double mu_;              //!< angle of scatter
-  double time_ {0.0};      //!< time in [s]
-  double time_last_ {0.0}; //!< previous time in [s]
-
-  // Other physical data
-  Position r_last_current_; //!< coordinates of the last collision or
-                            //!< reflective/periodic surface crossing for
-                            //!< current tallies
-  Position r_last_;         //!< previous coordinates
-  Direction u_last_;        //!< previous direction coordinates
-  double wgt_last_ {1.0};   //!< pre-collision particle weight
-
-  // What event took place
-  bool fission_ {false};  //!< did particle cause implicit fission
-  TallyEvent event_;      //!< scatter, absorption
-  int event_nuclide_;     //!< index in nuclides array
-  int event_mt_;          //!< reaction MT
-  int delayed_group_ {0}; //!< delayed group
-
-  // Post-collision physical data
-  int n_bank_ {0};        //!< number of fission sites banked
-  int n_bank_second_ {0}; //!< number of secondary particles banked
-  double wgt_bank_ {0.0}; //!< weight of fission sites banked
-  int n_delayed_bank_[MAX_DELAYED_GROUPS]; //!< number of delayed fission
-                                           //!< sites banked
-
-  // Indices for various arrays
-  int surface_ {0};        //!< index for surface particle is on
-  int cell_born_ {-1};     //!< index for cell particle was born in
-  int material_ {-1};      //!< index for current material
-  int material_last_ {-1}; //!< index for last material
+  vector<NuclideMicroXS> neutron_xs_;
+  vector<ElementMicroXS> photon_xs_;
+  MacroXS macro_xs_;
+  CacheDataMG mg_xs_cache_;
 
-  // Boundary information
-  BoundaryInfo boundary_;
+  ParticleType type_ {ParticleType::neutron};
+
+  double E_;
+  double E_last_;
+  int g_ {0};
+  int g_last_;
 
-  // Temperature of current cell
-  double sqrtkT_ {-1.0};     //!< sqrt(k_Boltzmann * temperature) in eV
-  double sqrtkT_last_ {0.0}; //!< last temperature
+  double wgt_ {1.0};
+  double mu_;
+  double time_ {0.0};
+  double time_last_ {0.0};
+  double wgt_last_ {1.0};
 
-  // Statistical data
-  int n_collision_ {0}; //!< number of collisions
+  bool fission_ {false};
+  TallyEvent event_;
+  int event_nuclide_;
+  int event_mt_;
+  int delayed_group_ {0};
+
+  int n_bank_ {0};
+  int n_bank_second_ {0};
+  double wgt_bank_ {0.0};
+  int n_delayed_bank_[MAX_DELAYED_GROUPS];
+
+  int cell_born_ {-1};
+
+  int n_collision_ {0};
 
-  // Track output
   bool write_track_ {false};
 
-  // Current PRNG state
-  uint64_t seeds_[N_STREAMS]; // current seeds
-  int stream_;                // current RNG stream
+  uint64_t seeds_[N_STREAMS];
+  int stream_;
 
-  // Secondary particle bank
   vector<SourceSite> secondary_bank_;
 
-  int64_t current_work_; // current work index
+  int64_t current_work_;
```

Comment:
Any particular reason the variable declaration comments were removed in this area?

Reply:
Yes, I moved comments to the accessors rather than the variables. IMO the external-facing interface should have the documentation, not the private implementation.